src/lock/mod.rs

```diff
@@ -77,10 +77,20 @@ pub fn config(ctx: &Context, config: Config) -> Result<LockedConfig> {
 
     // Create a map of unique `Source` to `Vec<Plugin>`
     let mut map = IndexMap::new();
+    let profile = std::env::var("SHELDON_PROFILE");
     for (index, plugin) in externals {
-        map.entry(plugin.source.clone())
-            .or_insert_with(|| Vec::with_capacity(1))
-            .push((index, plugin));
+        let profile_matches = match plugin.profiles {
+            None => true,
+            Some(ref profiles) => match profile {
+                Err(_) => false,
+                Ok(ref profile) => profiles.contains(profile),
+            },
+        };
+        if profile_matches {
```

Comment:
I think we should rather do this check before we install the sources in the closure below. 

```rust
            .map(|(source, plugins)| {
                let source_name = source.to_string();

                // filter plugins here, if len == 0, then log Skipped
                
                let source = source::lock(ctx, source)
                    .with_context(s!("failed to install source `{}`", source_name))?;
                    
                    ...
            })
```


Example output if the plugin source was skipped. E.g.

```
Loaded ~/.sheldon/plugins.toml
   Checked ~/.zsh/plugins
   Cloned https://github.com/sindresorhus/pure
   Skipped https://github.com/zsh-users/zsh-completions
Locked ~/.sheldon/plugins.lock
```